src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AmazonWebServicesCredentialsBinding.java

```diff
@@ -85,28 +94,66 @@ public String getSecretKeyVariable() {
         return secretKeyVariable;
     }
 
+    @DataBoundSetter
+    public void setRoleArn(String roleArn) {
+        this.roleArn = roleArn;
+    }
+
+    @DataBoundSetter
+    public void setRoleSessionName(String roleSessionName) {
+        this.roleSessionName = roleSessionName;
+    }
+
+    @DataBoundSetter
+    public void setRoleSessionDurationSeconds(int roleSessionDurationSeconds) {
+        this.roleSessionDurationSeconds = roleSessionDurationSeconds;
+    }
+
     @Override
     protected Class<AmazonWebServicesCredentials> type() {
         return AmazonWebServicesCredentials.class;
     }
 
     @Override
     public MultiEnvironment bind(@Nonnull Run<?, ?> build, FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
-        AWSCredentials credentials = getCredentials(build).getCredentials();
+        AWSCredentialsProvider provider = getCredentials(build);
+        if (!StringUtils.isEmpty(this.roleArn)) {
+            provider = this.assumeRoleProvider(provider);
+        }
+
+        AWSCredentials credentials = provider.getCredentials();
+
         Map<String,String> m = new HashMap<String,String>();
         m.put(accessKeyVariable, credentials.getAWSAccessKeyId());
         m.put(secretKeyVariable, credentials.getAWSSecretKey());
-
+    
         // If role has been assumed, STS requires AWS_SESSION_TOKEN variable set too.
         if(credentials instanceof AWSSessionCredentials) {
             m.put(SESSION_TOKEN_VARIABLE_NAME, ((AWSSessionCredentials) credentials).getSessionToken());
         }
         return new MultiEnvironment(m);
     }
 
+    private AWSSessionCredentialsProvider assumeRoleProvider(AWSCredentialsProvider baseProvider) {
+        AWSSecurityTokenService stsClient = AWSCredentialsUtils.buildStsClient(baseProvider);
+
+        String roleSessionName = StringUtils.defaultIfBlank(this.roleSessionName, "Jenkins");
+
+        STSAssumeRoleSessionCredentialsProvider.Builder assumeRoleProviderBuilder =
+                new STSAssumeRoleSessionCredentialsProvider.Builder(this.roleArn, roleSessionName)
+                .withStsClient(stsClient);
+
+        if (this.roleSessionDurationSeconds > 0) {
+            assumeRoleProviderBuilder = assumeRoleProviderBuilder
+                .withRoleSessionDurationSeconds(this.roleSessionDurationSeconds);
+        }
```

Comment:
We might have a problem here.
The setter requires a value between 900 and 3600, or it throws an `IllegalArgumentException`. But we are not checking this.
However, the `STSAssumeRoleSessionCredentialsProvider` constructor will use `DEFAULT_DURATION_SECONDS` (900) if the value is equal to 0.

I would suggest to either have a `doCheckX` method in the descriptor to validate that the duration is truly between 900 and 3600. You can even have the default value to `900`. This would not be used in pipeline setup.
Or you need to change the condition here.

(option 1)
```suggestion
                .withStsClient(stsClient)
                .withRoleSessionDurationSeconds(this.roleSessionDurationSeconds);
```

(option 2)
```suggestion
                .withStsClient(stsClient);

        if (this.roleSessionDurationSeconds >= 900 && this.roleSessionDurationSeconds <= 3600) {
            assumeRoleProviderBuilder = assumeRoleProviderBuilder
                .withRoleSessionDurationSeconds(this.roleSessionDurationSeconds);
        }
```

Reply:
My intent was to let the bounds just be controlled by the AWS SDK so that you don't have to keep the two in sync. (For example, a newer version of the SDK might change it so you can assume a role for up to 2 hours.) Similarly, I think it makes more sense to allow the default value to be chosen by the SDK, as it currently is.

Reply:
I understand. The problem is that the "check" will be done by the library which throws an exception. I'm not sure this is the most clear way to handle the situation for users.

Reply:
I'm a little confused. What exactly would we do in this code if the role session duration is out of bounds other than throw an exception ourselves?